plugin/clock_utils.h

```diff
@@ -0,0 +1,47 @@
+#ifndef GAZEBO_CLOCK_UTILS_H
+#define GAZEBO_CLOCK_UTILS_H
+
+#include <google/protobuf/duration.pb.h>
+#include <google/protobuf/timestamp.pb.h>
+#include <chrono>
+#include <cmath>
+
+// Convert a frequency into a duration
+template <typename period>
+struct Per;
+
+template <typename Unit, std::intmax_t num, std::intmax_t den>
+struct Per<std::chrono::duration<Unit, std::ratio<num, den>>> : public std::chrono::steady_clock::duration {
+    Per(int ticks)
+        : std::chrono::steady_clock::duration(std::lround(
+              (double(num) / double(ticks* den))
+              * (double(std::chrono::steady_clock::period::den) / double(std::chrono::steady_clock::period::num)))) {}
+};
+
+// Cast a google::protobuf::Timestamp to a std::chrono::steady_clock::time_point
+inline std::chrono::steady_clock::time_point time_point_cast(const ::google::protobuf::Timestamp& time_point) {
+    // Get our seconds and nanos in c++ land
+    auto seconds = std::chrono::seconds(time_point.seconds());
+    auto nanos   = std::chrono::nanoseconds(time_point.nanos());
+
+    // Make a timestamp out of the summation of them
+    return std::chrono::steady_clock::time_point(seconds + nanos);
+}
+
+// Cast a std::chrono::steady_clock::time_point to a google::protobuf::Timestamp
+inline ::google::protobuf::Timestamp time_point_cast(const std::chrono::steady_clock::time_point& t) {
```

Comment:
There is already a thing called `std::chrono::time_point_cast` do you think this will cause confusion?